include/seqan3/search/algorithm/detail/search.hpp

```diff
@@ -179,9 +184,12 @@ inline auto search_all(index_t const & index, queries_t & queries, configuration
     // delegate params: text_position (or cursor). we will withhold all hits of one query anyway to filter
     //                  duplicates. more efficient to call delegate once with one vector instead of calling
     //                  delegate for each hit separately at once.
+    using text_pos_t = std::conditional_t<index_t::is_collection,
```

Comment:
like here it seems a frequent pattern to ask such a thing. I am wondering if there is something like `index_traits<index_t>::...` If not, I think it seems a good thing to do. If relevant can you make a card for the strategy meeting.

Reply:
I'll add a card for now, but it's more a trait of the cursor than the index itself